src/renderers/shared/utils/ReactErrorUtils.js

```diff
@@ -155,33 +155,40 @@ if (__DEV__) {
       e,
       f,
     ) {
-      ReactErrorUtils._hasCaughtError = false;
-      ReactErrorUtils._caughtError = null;
-
       depth++;
-      const thisDepth = depth;
+
+      let error;
```

Comment:
We could also create our own error object with a helpful message explaining what happened. If there is browser flakiness, this would be more helpful to the developer than throwing undefined.